content/chainguard/chainguard-enforce/vulnerabilities/vulns-in-enforce/index.md

```diff
@@ -0,0 +1,63 @@
+---
+title: "Vulnerability Analysis with Chainguard Enforce"
+linktitle: "Vulnerability Analysis"
+description: "Reference tutorial outlining Chainguard Enforce Vulnerability analysis feature as well as the Enforce Console's Vulnerability filtering functionality."
+type: "article"
+date: 2023-06-27T08:48:45+00:00
+lastmod: 2023-06-27T08:48:45+00:00
+draft: false
+tags: ["Enforce", "Product", "Vulnerability", "Reference"]
+images: []
+weight: 050
+---
+
+Chainguard Enforce provides a comprehensive solution for monitoring your cloud workloads and container images. In particular, its vulnerability report generation and ingestion features enhance your ability to monitor the software components within your environment.
+
+Chainguard Enforce will automatically generate vulnerability reports using Grype on a daily basis. This ensures you are always aware of any new vulnerabilities that could affect your workloads.
+
+Enforce uses the previously generated or ingested SBOM in Enforce to generate a vulnerability report. This ensures that you always have a detailed view of the vulnerabilities in your images, without any additional effort on your part. Additionally, the Chainguard Enforce Console provides a handy interface that allows you to find and filter the severity of the discovered vulnerabilities on every cluster overseen by Enforce. This guide outlines the vulnerability analysis features in Enforce.
+
+## Chainguard Enforce automatically ingests Vulnerability reports out of the box
+
+When you run a workload with a supported Container Runtime (including EKS, GKE, and Cloud Run) monitored by Chainguard Enforce, it will first check if there are already vulnerability reports attached to the workload’s container images. We support signed cosign vulnerability reports (e.g. a vulnerability report uploaded using `cosign attest` as an in-toto attestation). Nowadays Chainguard Enforce supports Grype’s custom scan report format in JSON format. The result of a Grype scanning operation is included within the cosign vulnerability attestation specification. That attestation is supposed to be signed and include additional metadata about the scanner used to generate the vulnerability report. Enforce also checks the timestamp of the vulnerability attestation ignoring all reports whose age is older than 24 hours. If all these conditions are satisfied, Enforce will ingest that scan report.
```

Comment:
```suggestion
When you run a workload with a supported Container Runtime (including EKS, GKE, and Cloud Run) monitored by Chainguard Enforce, it will first check if there are already vulnerability reports attached to the workload’s container images. We support signed cosign vulnerability reports, for example a vulnerability report uploaded using `cosign attest` as an [in-toto attestation](https://in-toto.io/).

Chainguard Enforce supports Grype’s custom scan report format in JSON format. The result of a Grype scanning operation is included within the cosign vulnerability attestation specification. For Enforce to use an attestion, it should be signed and include additional metadata about the scanner used to generate the vulnerability report. Enforce also checks the timestamp of the vulnerability attestation ignoring all reports whose age is older than 24 hours. If all these conditions are satisfied, Enforce will ingest that scan report.
```